### ISSUE TITLE
[no-unused-properties] can't recognize $props.xxx used

### ISSUE THREAD

**Issue by levchak0910** [vuejs/eslint-plugin-vue#2344]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.55.0
- **eslint-plugin-vue version:** 9.19.2
- **Node version:** not applied, repro in playground
- **Operating System:** not applied, repro in playground

**Please show your full configuration:**

```json5
{
    "rules": {
        'vue/no-unused-properties': 'error',
    }
}
```

**What did you do?**	

```vue
<template>
  <div :class="$props.class"></div>
  <div>{{$props.for}}</div>
</template>

<script lang="ts" setup>
defineProps<{
  class: string
  for: number
}>()
</script>
```

**What did you expect to happen?**

props: `class` and `for` not reported as unused

**What actually happened?**

props: `class` and `for` reported as unused

**Repository to reproduce this issue**

[playground link](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxNz8GKwyAQgOFXkWEPXWg20KO4YR9hHyCXNJkWwU5kZiwU8d2rTbrscfz1UzPM64JgwSneYpgUh5GMcYu/GzuHSeR7hI/Ia5Sv1zjC4Ppa/7YNOe/9snIp7+j6f2AdZWYf1YSJrlXU6hhBTbHWBS+e8LcZLjf2dZE1ouzp2haqbA2l2xl5pDIcPpu/ifU8HIFTQAGb4Z6wp7VLlASXrr0LWX1rp3KEOLEg19/+6CPiBnQowZP2eytPzVthxA==)

**Bug description:**

When we want to define a prop name as a js keyword (for example: class, for, ...) it is required to use them as `$props.PROP`, because using simply as `PROP` will lead to [parsing error](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxNjjEKwzAMRa9iNLWQYOho3NAj9ABeTKwGg+sYSw6UkLtXSTp0EtL7evwVxjkgGLCM75I84+CyUjbERZkxeaK7g2M6GKyWs3Cr/8Ky0lhjYZV8niTNElWE3IrQgK+Y8VnnQnbdzYfLKOIa8+TyNlyuu/BUyAN0UFtCArPC0lDnuW+5EYa+iAQrx53dtg6Kr4RVqj/4U/AU9EgpZtY/tn0BRZFPmg==).

Another option is to use as `props.PROP` with props variable declaration

```vue
<template>
  <div :class="props.class"></div>
  <div>{{props.for}}</div>
</template>

<script lang="ts" setup>
const props = defineProps<{
  class: string
  for: number
}>()
</script>
```

but the rule doesn't support it too #2193

**Comment by levchak0910** [comment]
Thanks for clarifying status
I saw the #1285 issue, but it was not clear if fix is planned

**Comment by levchak0910** [comment]
@FloEdelmann I have a fix for this issue, but only for using `$props.PROP` in `<template>` (PROP will still be reported even if `$props.PROP` is used in `<script>`)

This fix is like a half-solution, so I would like to ask first: is it worth sending PR?

**Comment by FloEdelmann** [comment]
I guess that's already an improvement, so IMO it's definitely worth a PR.

**Comment by FloEdelmann** [comment]
Reopening this issue for only the specific issue of using `$props.xxx`.
